pkg/db/src/implementations/version-1.1/note.ts

```diff
@@ -250,12 +243,20 @@ class Note implements NoteInterface {
 	updateTransaction(transaction: VolumeStock): Promise<NoteInterface> {
 		// Create a safe copy of volume entries
 		const entries = [...this.entries];
-		const i = entries.findIndex(({ isbn }) => isbn === transaction.isbn);
+		// handle edge case where we have multiple books with the same isbn, but belonging to different warehouses
+		// match isbn only in case entry has no whId but transaction does (not undefined or empty string)
+		const i = entries.findIndex(
+			({ isbn, warehouseId }) =>
+				isbn === transaction.isbn &&
+				(warehouseId === versionId(transaction.warehouseId || '') || (warehouseId === versionId('') && transaction.warehouseId))
+		);
+
+		const versionedTransaction = { ...transaction, warehouseId: versionId(transaction.warehouseId || '') };
```

Comment:
Maybe do this before trying to match the entry...less `warehouseId` conversions on the fly. 
Also: we don't want `v1/` as `warehouseId`